From f030016072e61eaa45ced9de065d91cd5f7741cd Mon Sep 17 00:00:00 2001
From: Ole Streicher <ole@aip.de>
Date: Fri, 23 Sep 2016 09:40:09 +0200
Subject: [PATCH] Unconditionally use `int` as data type for FitsLong. This is
 currently the case on all useful platforms.

This commit fixes some problems on some 64-bit architectures, namely arm64 and
s390x which used the wrong typedef here because they were not explicitely
listed.
---
 casa/aipsxtype.h | 11 +++--------
 fits/FITS/fits.h |  9 ++-------
 2 files changed, 5 insertions(+), 15 deletions(-)

diff --git a/casa/aipsxtype.h b/casa/aipsxtype.h
index aff1f4a..97bfdbf 100644
--- a/casa/aipsxtype.h
+++ b/casa/aipsxtype.h
@@ -38,14 +38,9 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
 typedef long long Int64;
 typedef unsigned long long uInt64;
 
-//# All FITS code seems to assume longs are 4 bytes. Take care of machines 
-//# for which this isn't true here by defining FitsLong to be the 4 byte int.
-//# Use FitsLong instead of long in the FITS code where it matters.
-#if (defined(AIPS_ALPHA) || defined(AIPS_SGI) || defined(__x86_64__))
-    typedef int FitsLong;
-#else
-    typedef long FitsLong;
-#endif 
+//# All FITS code seems to assume longs are 4 bytes. Currently
+//# this corresponds to an "int" on all useful platforms.
+typedef int FitsLong;
 
 } //# NAMESPACE CASACORE - END
 
diff --git a/fits/FITS/fits.h b/fits/FITS/fits.h
index 12a371a..8906efe 100644
--- a/fits/FITS/fits.h
+++ b/fits/FITS/fits.h
@@ -39,14 +39,9 @@
 
 namespace casacore { //# NAMESPACE CASACORE - BEGIN
 
-//# All FITS code seems to assume longs are 4 bytes. To take care of machines 
-//# for which this isn't true use FitsLong instead of Long in the FITS code
-//# where it matters.
-# if (defined(__alpha) || defined(__sgi) || defined(__x86_64__))
+//# All FITS code seems to assume longs are 4 bytes. Currently
+//# this corresponds to an "int" on all useful platforms.
     typedef Int FitsLong;
-# else
-    typedef Long FitsLong;
-# endif 
 //# recovered by GYL
 
 //# Forward declarations
